namespace/Tiltfile

```diff
@@ -0,0 +1,71 @@
+# -*- mode: Python -*-
+
+def namespace_yaml(name):
+  """Returns YAML for a namespace
+
+  Args:
+    name: The namespace name. Currently not validated.
+
+  Returns:
+    The namespace YAML as a blob
+  """
+
+  return blob("""apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+""" % name)
+
+def namespace_create(name):
+  """Returns YAML for a namespace
+
+  Args:
+    name: The namespace name. Currently not validated.
+  """
+  k8s_yaml(namespace_yaml(name))
+
+def namespace_inject(x, ns):
+  """Takes K8s yaml, sets its namespace to `ns`, and returns it as a blob.
+
+  This modifies the yaml in two ways:
+  1. Sets .metadata.namespace to `ns`
+  2. Sets ..template.metadata.namespace to `ns`
+     This ensures the namespace in, e.g., Deployment Pod Template Specs is
+     set, but might have false positives if you have a CRD with some other
+     element named 'template'.
+
+  Args:
+    x: K8s yaml. Either a filename (string) or the yaml itself (Blob)
+    ns: The namespace to set the K8s objects to.
+
+  Returns:
+    Blob containing the K8s object as yaml, with namespaces set to `ns`.
+  """
+  return _mutate_yaml(x, lambda o: _set_k8s_yaml_namespace(o, ns))
+
+
+def _mutate_yaml(x, f):
+  if type(x) == 'string':
+    objects = read_yaml_stream(x)
+  elif type(x) == 'blob':
+    objects = decode_yaml_stream(x)
+  else:
+    fail('only takes string or blob, got: %s' % type(x))
+
+  return encode_yaml_stream([f(o) for o in objects])
+
+def _set_k8s_yaml_namespace(o, ns):
+  o['metadata']['namespace'] = ns
```

Comment:
Ideally we'd only do this to namespace-scoped objects. That seems hard, so maybe just a caveat in the docs not to use this on cluster-scoped objects?

Reply:
or, hmm, I just tried creating a namespace with metadata.namespace set, and kubectl apply simply ignored the namespace, so that's probably fine? I couldn't find official documentation that k8s ignores namespaces on cluster-scoped objects, but some related discussion [here](https://github.com/kubernetes-sigs/kustomize/issues/552).

Reply:
added a caveat